lektor/builder.py

```diff
@@ -487,16 +489,14 @@ def checksum(self):
         try:
             h = hashlib.sha1()
             if os.path.isdir(self.filename):
-                h.update('DIR\x00')
+                h.update('DIR\x00'.encode('utf-8'))
                 for filename in sorted(os.listdir(self.filename)):
                     if self.env.is_uninteresting_source_name(filename):
                         continue
-                    if isinstance(filename, unicode):
-                        filename = filename.encode('utf-8')
-                    h.update(filename)
+                    h.update(filename.encode('utf-8'))
```

Comment:
Filenames cannot be unicode on 2.x and are not guarnateed to be on 3.x either so you will need to do something else where where both bytes and unicode can be accepted.  The 2.x code already did this correctly.